src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -626,7 +626,7 @@ export class DataViewsService {
 
       // If allowNoIndex, only update field list if field caps finds fields. To support
       // beats creating index pattern and dashboard before docs
-      if (!options.allowNoIndex || (newFields && newFields.length > 5)) {
+      if (!options.allowNoIndex || (newFields && newFields.length > 4)) {
```

Comment:
I suspect this fixes a test ... but curious... why?

Reply:
I'm going to close this PR is favor of https://github.com/elastic/kibana/pull/152059 although I'm going to move some test fixes over.

the `allowNoIndices` data views attribute changes some field list loading behavior. It was a hack added to prevent the display of errors for the ingest team before data was present. It was always a terrible hack, I think it was added after feature freeze, one of those 'we need to do this so we can ship software, we'll think this through later'. 

The code that uses the dataViews.allowNoIndices attribute should be examined for removal. 

The PR I opened only changes the `allowNoIndices` param for field list loading.